www/front_src/src/Resources/Graph/Performance/AnomalyDetection/index.cypress.spec.tsx

```diff
@@ -0,0 +1,209 @@
+
+import { act, renderHook } from '@testing-library/react-hooks/dom';
+import { Provider, useAtom } from 'jotai';
+import { BrowserRouter as Router } from 'react-router-dom';
+
+import { platformVersionsAtom } from '../../../../Main/atoms/platformVersionsAtom';
+import { authorizedFilterByModules } from '../../../Filter/Criterias/models';
+import { storedFilterAtom } from '../../../Filter/filterAtoms';
+import { allFilter } from '../../../Filter/models';
+import Resources from '../../../index';
+import { enabledAutorefreshAtom } from '../../../Listing/listingAtoms';
+import {
+  labelCancel,
+  labelClose,
+  labelDisplayEvents,
+  labelGraph,
+  labelLast31Days,
+  labelLast7Days,
+  labelLastDay,
+  labelMenageEnvelope,
+  labelMenageEnvelopeSubTitle,
+  labelUseDefaultValue,
+} from '../../../translatedLabels';
+
+const installedModules = {
+  modules: {
+    'centreon-anomaly-detection': {
+      fix: '0-beta',
+      major: '22',
+      minor: '10',
+      version: '22.10.0-beta.1',
+    },
+  },
+  web: {},
+  widgets: [],
+};
+
+const moduleName = 'centreon-anomaly-detection';
+
+interface Search {
+  type: string;
+}
+
+const filtersToBeDisplayedInTypeMenu = Object.values(
+  authorizedFilterByModules[moduleName],
+);
+
+const filtersToBeDisplayedInSearchBar = Object.keys(
+  authorizedFilterByModules[moduleName],
+);
+
+const searchWords = filtersToBeDisplayedInSearchBar.reduce(
+  (prevValue, currentValue: string): Search => {
+    const value = prevValue.type;
+    const searchKeyWords = {
+      type: value ? `${value},${currentValue}` : `${currentValue}`,
+    };
+
+    return { ...prevValue, ...searchKeyWords };
+  },
+  { type: '' },
+);
+
+document.getElementById('cy-root').style = 'min-height:750px;display:flex';
+
+describe('Anomaly detection ', () => {
+  beforeEach(() => {
+    cy.viewport(1200, 750);
+    cy.fixture('resources/resourceListing.json').as('listResource');
+    cy.fixture('resources/userFilter.json').as('userFilter');
+    cy.fixture('resources/detailsAnomalyDetection.json').as(
+      'detailsAnomalyDetection',
+    );
+    cy.fixture('resources/performanceGraphAnomalyDetection.json').as(
+      'graphAnomalyDetection',
+    );
+    cy.server();
+    cy.route('GET', '**/resources?*', '@listResource').as('getResourceList');
+    cy.route('GET', '**/events-view?*', '@userFilter').as('filter');
+    cy.route(
+      'GET',
+      '**/resources/anomaly-detection/1',
+      '@detailsAnomalyDetection',
+    );
+    cy.route('GET', '**/performance?*', '@graphAnomalyDetection').as(
+      'getGraphDataAnomalyDetection',
+    );
+
+    const storedFilter = renderHook(() => useAtom(storedFilterAtom));
+
+    act(() => {
+      storedFilter.result.current[1](allFilter);
+    });
+
+    cy.mount(
+      <Provider
+        initialValues={[
+          [platformVersionsAtom, installedModules],
+          [enabledAutorefreshAtom, false],
+        ]}
+      >
+        <Router>
+          <Resources />
```

Comment:
I'm not sure this a good way to do component testing. I think it would be better to test component by component. For instance, it's better to test Filters component only then the anomaly detection graph component :)